versioneer.py

```diff
@@ -1537,7 +1608,7 @@ def pep440_split_post(ver):
     return vc[0], int(vc[1] or 0) if len(vc) == 2 else None
 
 
-def render_pep440_pre(pieces):
+def render_pep440_pre(pieces: Dict[str, Any]) -> str:
     """TAG[.postN.devDISTANCE] -- No -dirty.
```

Comment:
Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)